src/View/SSViewer_DataPresenter.php

```diff
@@ -117,85 +130,29 @@ protected function createCallableArray(&$extraArray, $interfaceToQuery, $variabl
-                        $source = self::$iteratorProperties[$property];
-                        if ($this->itemIterator) {
-                            // Set the current iterator position and total (the object instance is the first item in
-                            // the callable array)
-                            $source['implementer']->iteratorProperties(
-                                $this->itemIterator->key(),
-                                $this->itemIteratorTotal
-                            );
-                        } else {
-                            // If we don't actually have an iterator at the moment, act like a list of length 1
-                            $source['implementer']->iteratorProperties(0, 1);
-                        }
-                    } // And finally for global overrides
-                    else {
-                        if (array_key_exists($property, self::$globalProperties)) {
-                            $source = self::$globalProperties[$property];  //get the method call
-                        }
-                    }
-                }
-            }
+        // Get source for this value
+        $source = $this->getValueSource($property);
+        if (!$source) {
+            return null;
         }
 
-        if ($source) {
-            $res = array();
-
-            // Look up the value - either from a callable, or from a directly provided value
-            if (isset($source['callable'])) {
-                $res['value'] = call_user_func_array($source['callable'], $params);
-            } elseif (isset($source['value'])) {
-                $res['value'] = $source['value'];
-            } else {
-                throw new InvalidArgumentException("Injected property $property does't have a value or callable " .
-                    "value source provided");
-            }
-
-            // If we want to provide a casted object, look up what type object to use
-            if ($cast) {
-                // If the handler returns an object, then we don't need to cast.
-                if (is_object($res['value'])) {
-                    $res['obj'] = $res['value'];
-                } else {
-                    // Get the object to cast as
-                    $casting = isset($source['casting']) ? $source['casting'] : null;
-
-                    // If not provided, use default
-                    if (!$casting) {
-                        $casting = ViewableData::config()->uninherited('default_cast');
-                    }
-
-                    $obj = Injector::inst()->get($casting, false, array($property));
-                    $obj->setValue($res['value']);
-
-                    $res['obj'] = $obj;
-                }
-            }
+        // Look up the value - either from a callable, or from a directly provided value
+        $res = [];
+        if (isset($source['callable'])) {
+            $res['value'] = call_user_func_array($source['callable'], $params);
```

Comment:
Appreciate you didn’t write this, but we can now use (the apparently slightly faster) argument unpacking:

```php
$callable = $source['callable'];
$res['value'] = $callable(...$params);
```

Reply:
Won't work if using `[$this, 'methodName']` approach I think. I use call_user_func_array for legacy purposes.